seahorn/lib/proof_allocators.c

```diff
@@ -40,17 +40,49 @@ static void s_free_allocator(struct aws_allocator *allocator, void *ptr) {
   free(ptr);
 }
 
+static void *s_realloc_allocator(struct aws_allocator *allocator, void *ptr, size_t oldsize, size_t newsize) {
+  (void)allocator;
+  (void)oldsize;
+  AWS_FATAL_PRECONDITION(newsize);
+
+  if (newsize <= oldsize) {
+    return ptr;
+  }
+
+  /* newsize is > oldsize, need more memory */
+  void *new_mem = s_malloc_allocator(allocator, newsize);
+  // AWS_PANIC_OOM(new_mem, "Unhandled OOM encountered in s_malloc_allocator");
```

Comment:
Can realloc return a `NULL` ptr? If so then let's just remove the commented line.

Reply:
Yes. `realloc` can fail and return a null pointer.